finality-provider/service/fp_instance.go

```diff
@@ -98,10 +97,17 @@ func (fp *FinalityProviderInstance) Start() error {
 		return fmt.Errorf("the finality-provider instance %s is already started", fp.GetBtcPkHex())
 	}
 
+	if fp.IsJailed() {
+		return fmt.Errorf("%w: %s", ErrFinalityProviderJailed, fp.GetBtcPkHex())
+	}
+
 	fp.logger.Info("Starting finality-provider instance", zap.String("pk", fp.GetBtcPkHex()))
 
 	startHeight, err := fp.bootstrap()
 	if err != nil {
+		if errors.Is(err, ErrFinalityProviderJailed) {
+			fp.MustSetStatus(proto.FinalityProviderStatus_JAILED)
+		}
```

Comment:
Do you think this should be done here or inside `bootstrap()`?

Reply:
Good point. Moved the error handler to `bootstrap()`